library/src/main/java/software/amazon/location/auth/utils/CryptoUtils.kt

```diff
@@ -0,0 +1,75 @@
+package software.amazon.location.auth.utils
+
+import java.io.UnsupportedEncodingException
+import java.net.URLEncoder
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import software.amazon.location.auth.utils.Constants.DEFAULT_ENCODING
+
+internal const val HASHING_ALGORITHM = "SHA-256"
+internal const val MAC_ALGORITHM = "HmacSHA256"
+internal const val ENCODED_CHARACTER_REGEX = "%[0-9A-Fa-f]{2}"
+
+internal fun hash(value: String): String {
+    val bytes = value.toByteArray()
+    val md = MessageDigest.getInstance(HASHING_ALGORITHM)
+    val digest = md.digest(bytes)
+    return digest.toHexString()
+}
+
+@Throws(Exception::class)
+internal fun hmacSha256(key: ByteArray, data: String): ByteArray {
```

Comment:
Do we have to write this code ourselves, or is there anything from the AWS Kotlin SDK that can be used?

Reply:
I didn't find any inbuilt code in AWS Kotlin SDK to do this encrypt? But I can see [here (Java Sdk)](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/auth/AWS4Signer.java).